cmd/controller-manager/app/controllermanager.go

```diff
@@ -46,22 +50,35 @@ var knownControllers = map[string]startControllerFunc{
 
 var controllersDisabledByDefault = sets.New(MonitorControllerName)
 
-// startControllerFunc is responsible for constructing and starting a controller. startControllerFunc should be
-// asynchronous and an error is only returned if we fail to start the controller.
-type startControllerFunc func(ctx context.Context, controllerCtx *controllercontext.Context) error
-
 // Run starts the controller manager according to the given options.
 func Run(ctx context.Context, opts *options.Options) {
 	controllerCtx, err := createControllerContext(opts)
 	if err != nil {
 		klog.Fatalf("Error creating controller context: %v", err)
 	}
 
+	if opts.EnableProfiling {
+		go func() {
+			if err := http.ListenAndServe("0.0.0.0:6060", nil); err != nil {
+				klog.Errorf("Failed to start pprof server: %v", err)
+			}
+		}()
+	}
+
+	handler := healthcheck.NewMutableHealthCheckHandler()
+	handler.AddLivezChecker("ping", healthz.Ping)
+
+	var healthzAdaptor *leaderelection.HealthzAdaptor
+	if opts.EnableLeaderElect {
+		healthzAdaptor = &leaderelection.HealthzAdaptor{}
```

Comment:
Initializing `HealthzAdaptor{}` like this implies setting `maxTolerableExpiredLease` to 0 when checking. Is this intended?

Reply:
no, fixed it